lib/mix/test/mix/tasks/escript_test.exs

```diff
@@ -94,6 +106,51 @@ defmodule Mix.Tasks.EscriptTest do
       Mix.Tasks.Escript.Build.run []
       assert_received {:mix_shell, :info, ["Generated escript escriptest with MIX_ENV=dev"]}
       assert System.cmd("escript", ["escriptest"]) == {"FROM CONFIG\n", 0}
+      assert count_abstract_code("escriptest") == 0
+    end
+  end
+
+  test "generate escript with debug information" do
+    Mix.Project.push EscriptWithDebugInfo
+
+    in_fixture "escripttest", fn ->
+      Mix.Tasks.Escript.Build.run []
+      assert_received {:mix_shell, :info, ["Generated escript escripttestwithdebuginfo with MIX_ENV=dev"]}
+      assert System.cmd("escript", ["escripttestwithdebuginfo"]) == {"TEST\n", 0}
+      assert count_abstract_code("escripttestwithdebuginfo") > 0
+
+      Mix.Tasks.Escript.Build.run []
+      refute_received {:mix_shell, :info, ["Generated escript escripttestwithdebuginfo with MIX_ENV=dev"]}
+    end
+  end
+
+  defp count_abstract_code(escript_filename) do
+    for {_filename, beam} <- get_beams(escript_filename) do
```

Comment:
Thanks; fixed to use `Enum.map` instead of `for`.  I still have to reject the `nil`s afterwards.